sycl/include/sycl/ext/oneapi/backend/hip.hpp

```diff
@@ -16,16 +16,33 @@ inline namespace _V1 {
 template <>
 inline backend_return_t<backend::ext_oneapi_hip, device>
 get_native<backend::ext_oneapi_hip, device>(const device &Obj) {
-  // TODO use SYCL 2020 exception when implemented
   if (Obj.get_backend() != backend::ext_oneapi_hip) {
-    throw sycl::runtime_error(errc::backend_mismatch, "Backends mismatch",
-                              PI_ERROR_INVALID_OPERATION);
+    throw sycl::exception(make_error_code(errc::backend_mismatch),
+                          "Backends mismatch");
   }
   // HIP uses a 32-bit int instead of an opaque pointer like other backends,
   // so we need a specialization with static_cast instead of reinterpret_cast.
   return static_cast<backend_return_t<backend::ext_oneapi_hip, device>>(
       Obj.getNative());
 }
 
+template <>
+inline device make_device<backend::ext_oneapi_hip>(
```

Comment:
Why is this code unique to HIP?

Reply:
There are a hierarchy of reasons. Firstly if we want to unique devices (https://github.com/intel/llvm/issues/6055) like in the cuda backend, https://github.com/intel/llvm/pull/7550 , then the code is unique to HIP (since the cuda implementation is in the experimental namespace). If we chose to not unique the device, then we would still need the code to be unique to hip, since, like the cuda backend, we would need to `static_cast` between `pi_native_handle` and `backend_input_t<backend::ext_oneapi_cuda, device>` since the default `make_device` uses reinterpret case which does not work.